### DISCUSSION TITLE
Livewire 3 + AlpineJS : "Uncaught TypeError: Illegal invocation"

### DISCUSSION THREAD

**Discussion by jornatf** [livewire/livewire#7009] [1. Help]
I'm using **Livewire 3 + AlpineJS**. When I initialize the page with all my custom Alpine functions (Alpine.data(...)), I don't see any errors in my console. However, as I navigate, I receive a lot of Alpine-related errors, and Livewire is telling me that Alpine functions or variables cannot be found.

I'm using "SPA" navigation (with the `wire:navigate` property). It seems that once Livewire is loaded for the first time, it doesn't retain the Alpine variables in memory.

Here is a screenshot of the console:

![image](https://github.com/livewire/livewire/assets/92311176/0ba78484-4ef9-4ce3-a83a-f9a1c213b889)

In my `app.js` file:

```js
document.addEventListener('alpine:init', () => {
    Alpine.data('abilities', abilities)
    Alpine.data('accordion', accordion)
    Alpine.data('datatable', datatable)
    Alpine.data('dragAndDrop', dragAndDrop)
   ...
})
```

Thanks for any helps.

Jordan

**Comment by devoided** [comment]
Did you ever resolve this? @jornatf

**Reply by jornatf** [reply]
@devoided Nop, I finally gave up on SPA navigation with Livewire.

**Reply by devoided** [reply]
Why did you close as resolved? To my understanding this is an ongoing bug

**Comment by apavliukov** [comment]
I fixed this issue with changing `Alpine.data(...)` to `window.Alpine.data(...)`.

According to https://livewire.laravel.com/docs/upgrading#alpinejs I also changed `@livewireScripts` to `@livewireScriptConfig` and `Alpine.start()` to `Livewire.start()`. So no need to import Alpine in JS.

**Comment by giltotherescue** [comment]
This is happening to me, and I figured out how to reproduce it.

In my case, this bug happens when I am streaming content to the view using wire:stream. The stream takes about 20 seconds to complete. If the user navigates to a different livewire view (using wire:navigate) during this time, everything is fine. However, if they navigate back to the original view where the stream occurred, the bug happens. 

Ideally I would like to cancel the stream if the user navigates to a different view. However, I haven't found a way to do this yet. And there is an [open discussion](https://github.com/livewire/livewire/discussions/7882) regarding this issue. 

Until there is a way to cancel a stream, I am exploring an approach to block navigation until the stream completes. To do this I will implement a cancellable alpine:navigate [using this patch which was just released today](https://github.com/livewire/livewire/pull/8093). However, this isn't ideal since it disrupts the user experience.

**Comment by SeyamMs** [comment]
I know this might be a weird solution but it worked.

just wrap the tag that has the `x-show` directive with a template tag with `x-if="true"`
